### ISSUE TITLE
Web login unusable

### ISSUE THREAD

**Issue by Rajani-Bp** [AIKEJI-AU/Xero.OAuth2.Docs#6]
Hi there
We are integrating your app in to our WinForms application (.net framework 4.6.2) and running in to issues with web login screen. When launched first time (from cold start), either it opens black screen or whole login screen looks out of place! It is difficult to enter xero login and password.

But we close the screen and re-open and all is fine. But it is annoying our Users any advise ?

**Comment by Rajani-Bp** [comment]
Screen shot , Login screen with the issue

![image](https://user-images.githubusercontent.com/91445091/134839889-a14bec74-285f-447c-84f7-33bd4db8aaad.png)

When launching second time, no issue. See how login screen renders correctly

![image](https://user-images.githubusercontent.com/91445091/134839956-5a0fa82e-b7dd-4a3e-b390-f9eac56f15ff.png)

**Comment by alice-aikeji** [comment]
Hi Rajani,

Which version of the library is being used when this issue happens? Also, what is your HighDPI setting?
Could you try upgrade to the latest version (if not already) and configure the library with the following settings and let me know if the issue persists?
HighDPI: true
FormWidth: 1024
FormHeight: 768

Thanks

**Comment by Rajani-Bp** [comment]
Alice

As per NuGet we are already using latest version i.e. 1.5.2

Yes, we already tried setting High DPI and setting it to true and makes it even worse! It makes our whole application tiny and unusable

Not sure screenshot below explains better but here you go

***@***.***

Our application behind which was opened full screen now became smaller than web login screen! This is with your suggested changes.

And like I mentioned in my original email , opening the login screen second time resolves itself.
BTW this is our current code for Token Helper

***@***.***

Thank for your email and we have impending release with Xero shutting down oAuth1 this month and appreciate a solution sooner than later.

Regards
Rajani

[Bp Logo]

Rajani Puvvula
Development Leader
Hamilton Operations Centre L2, 130 Grantham Street,
Hamilton New Zealand 3204 | 0800 40 1111
[Bp Website]<http://www.bpsoftware.net/>

**Comment by alice-aikeji** [comment]
Hi Rajani,

Thanks for sharing additional information and screenshots.

I shared the recommended settings because that's what our test app uses and the result has been consistent. Apologies that it made your app looked so out of place! Forget about my recommendation.

I am having trouble replicating the black screen issue at my end.

On further research, it seems that the cold boot black screen issue for Chromium Embedded Browsers has been around for a while. I read forum posts dating back to 2014 and recent posts suggesting that it is broken again since v84. This makes sense for what you are experiencing. The CEF version the library uses is v85.
This also concerns me because if it is a bug in the Chromium Browser, I doubt changing code in our library will fix it.

Although I'm happy to hear insight from you and other users regarding this. And also wondering how many users are experiencing this issue.

It is probably not the answer you want to hear but I'm at my wit's end. : (

**Comment by Rajani-Bp** [comment]
Hi Alice

Thanks for conforming that, Alice. I had been checking our code and setup thinking it was the problem. 
Anyway, It's not everyone about half experiencing this issue (black screen). So far we've not noticed any pattern/logic for this yet.  What is the solution for this? Keep checking for NuGet updates?

Note, All our clients are on same version.

One other issue and don't know if its related but so far only one Client reported it, issue is with CefSharp

**_Could not load file or assembly 'CefSharp.Core.dll' or one of its dependencies. The specified module could not be found._**


We distribute entire x86 folder and it is still complains about this missing DLL. I can see that file exists in the installation directory! Any thoughts  ?

The x86 I was referring to is this,
![image](https://user-images.githubusercontent.com/91445091/135777858-f5446890-ee70-4168-b68b-adc68f7de830.png)

**Comment by Rajani-Bp** [comment]
Please note, x86 is sub folder of our main installation directory where our application exists along with Aikeji DLLs

**Comment by Rajani-Bp** [comment]
Hi Alice

re this error ,
"Could not load file or assembly 'CefSharp.Core.dll' or one of its dependencies. The specified module could not be found."

We finally managed to find the what the issue was, It is Microsoft VC++ was missing on client machine and as soon as we installed it problem went away. 

It appears it is one of requirements for CefSharp. May I suggest , your wiki reflects this so save others time in the future?
